### DISCUSSION TITLE
Failover Rates - Custom Rate Scripting to Allow for multiple sources so invoices generate during provider outages

### DISCUSSION THREAD

**Discussion by astupidmoose** [btcpayserver/btcpayserver#2267] [Ideas / Feature Requests]
So with all the major exchanges running on windows 95 era servers over the last few weeks and crashing with anything larger than a 2% spike in price I've noticed that quite often our own site will go down because we aren't able to generate rates from the providers we have picked. 

We use the custom rate scripting, and as far as I know I can only define one coin per one exchange. I'd like to know if there is a way, or the possibility to add, the ability to use multiple rate providers in a priority type instance. 

For instance, if you use coinbase for your rate, but have binance set as a failover, then when Coinbase inevitably crashes you are still able to generate invoices using binance, until someone at coinbase turns the power back on at which point it would switch back over. 

Would appreciate ideas and thoughts.

**Comment by pirus111** [comment]
I would like to signal boost this request.

Maybe once every 1-2 months a rate provider will be temporarily unavailable.
During this period invoice creation fails.

The proposed solution is to allow either custom rate scripting, or (more preferably) the ability to select rate provider failovers.
e.g. You would pick rate providers as follows: Coinbase Pro > Coingecko > Kraken

As such, the odds of invoice generation failing due to this issue will be greatly reduced.

**Comment by pavlenex** [comment]
Logged as an issue, so will close this one let's keep conversation there.
